assets/js/base/context/cart-checkout/event_emit/emitters.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * Emits events on registered observers for the provided type and passes along
+ * the provided data.
+ *
+ * This event emitter will silently catch promise errors, but doesn't care
+ * otherwise if any errors are caused by observers. So events that do care
+ * should use `emitEventWithAbort` instead.
+ *
+ * @param {Object} observers The registered observers to omit to.
+ * @param {string} eventType The event type being emitted.
+ * @param {*}      data      Data passed along to the observer when it is
+ *                           invoked.
+ *
+ * @return {Promise} A promise that resolves to true after all observers have
+ *                   executed.
+ */
+export const emitEvent = async ( observers, eventType, data ) => {
+	const observersByType = observers[ eventType ] || [];
+	for ( let i = 0; i < observersByType.length; i++ ) {
+		try {
+			await Promise.resolve( observersByType[ i ]( data ) );
```

Comment:
`Promise.resolve` is used here to ensure that every observer is handled like a promise.